spec/controllers/v2/reports_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe V2::ReportsController do
+  before do
+    stub_rbac_permissions(
+      Rbac::INVENTORY_HOSTS_READ,
+      Rbac::SYSTEM_READ,
+      Rbac::REPORT_READ
+    )
+  end
+
+  let(:attributes) do
+    {
+      title: :title,
+      os_major_version: :os_major_version,
+      ref_id: :ref_id,
+      description: :description,
+      profile_title: :profile_title,
+      business_objective: :business_objective,
+      system_count: -> { 0 }, # TODO
+      # TODO: compliant_system_count: -> { 0 },
+      # TODO: test_result_system_count: -> { 0 },
+      # TODO: unsupported_system_count: -> { 0 },
+      compliance_threshold: :compliance_threshold
+    }
+  end
+
+  let(:current_user) { FactoryBot.create(:v2_user, :with_cert_auth) }
+  let(:rbac_allowed?) { true }
+
+  before do
+    request.headers['X-RH-IDENTITY'] = current_user.account.identity_header.raw
+    allow(StrongerParameters::InvalidValue).to receive(:new) { |value, _| value.to_sym }
+    allow(controller).to receive(:rbac_allowed?).and_return(rbac_allowed?)
+  end
+
+  context '/reports' do
+    describe 'GET index' do
+      let(:extra_params) { { account: current_user.account } }
+      let(:parents) { nil }
+
+      # TODO: only creates duplicated records
+      let(:items) do
+        item_count.times.map do
+          test_result = FactoryBot.create(
```

Comment:
Create `test_result` with inaccessible host, maybe outside of `items`.

Multiple test_results *for various systems* should be generated.